src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -472,6 +495,55 @@ private static String addPrefix(String server) {
         else return "ldap://" + server;
     }
 
+    private String generateId() {
+        return generateId(server, rootDN, userSearchBase, userSearch);
+    }
+
+    @Restricted(NoExternalUse.class)
+    static String generateId(String serverUrl, String rootDN, String userSearchBase, String userSearch) {
+        final MessageDigest digest = DigestUtils.getMd5Digest();
+        digest.update(normalizeServer(serverUrl).getBytes(Charsets.UTF_8));
+        if (StringUtils.isNotBlank(rootDN)) {
+            digest.update(rootDN.getBytes(Charsets.UTF_8)); //Should have been inferred in the constructor if needed
+        } else {
+            digest.update(new byte[]{0});
+        }
+        if (StringUtils.isNotBlank(userSearchBase)) {
```

Comment:
so these are three equivalent configurations:

* root DN `dc=example,dc=com` and user search base `dc=Users`
* root DN `dc=com` and user search base `dc=Users,dc=example`
* empty root DN (with infer root DN unchecked) and user search base `dc=Users,dc=example,dc=com`

You do not generate the same ID for these three configurations, :bug: